batch/batch/driver/instance_collection/pool.py

```diff
@@ -312,10 +321,10 @@ async def ready_cores_mcpu_from_estimated_job_queue(self):
         for user, share in user_share.items():
             user_job_query = f'''
 (
-SELECT CAST(COALESCE(SUM(cores_mcpu), 0) AS SIGNED) AS ready_cores_mcpu
+SELECT *, ROW_NUMBER() OVER (ORDER BY batch_id ASC, always_run DESC, job_id ASC) DIV {share} AS scheduling_iteration
```

Comment:
Can you give me a quick explanation of what this is doing? I haven't used over before.